contracts/periphery/KSZap.sol

```diff
@@ -132,27 +174,36 @@ contract ZapInV2 {
         uint8 v,
         bytes32 r,
         bytes32 s
-    ) external ensure(deadline) returns (uint256 amountOut) {
-        uint256 value = approveMax ? uint256(-1) : liquidity;
-        IERC20Permit(pool).permit(msg.sender, address(this), value, deadline, v, r, s);
-        amountOut = _zapOut(tokenIn, tokenOut, liquidity, pool, minTokenOut);
+    ) external ensure(deadline) whitelistedFactory(factory) returns (uint256 amountOut) {
```

Comment:
duplicated check here for whitelisted factory sine _zapOut already checked, similar for other functions.
We should just check in the internal function

Reply:
yeah it's duplicate but I think we should keep the check at the beginning of function so that user would not lose a lot of gas since if it's wrong factory then it reverted right away instead of calculate stuffs then revert.